src/header.jsx

```diff
@@ -115,3 +119,47 @@ const ViewSelector = ({ isGrid, setIsGrid, sortBy, setSortBy }) => {
         </Select>
     );
 };
+
+const UploadButton = ({ currentDir, files }) => {
+    const ref = useRef();
+
+    const handleClick = () => {
+        ref.current.click();
+    };
+
+    const onUpload = e => {
+        const uploadedFile = e.target.files[0];
+
+        let fileName = uploadedFile.name;
+
+        // eslint-disable-next-line no-loop-func
+        while (files.some(f => f.name === fileName)) {
+            fileName += " (1)";
+        }
+
+        const reader = new FileReader();
+        reader.readAsArrayBuffer(uploadedFile, "UTF-8");
```

Comment:
This loads the *whole* file into memory, which is a no-go for large files (think videos or VM images, which can easily be multiple GB). This needs to be chunked, i.e. read blocks (of maybe 100 kB or so) in a loop.

Reply:
A progressbar is also required if we do large uploads. which can be calculated based on chunks handled.